src/frontends/onnx/frontend/src/core/model.cpp

```diff
@@ -27,21 +27,26 @@ std::int64_t get_opset_version(const ONNX_NAMESPACE::ModelProto& model_proto, co
 }
 
 Model::Model(std::shared_ptr<ONNX_NAMESPACE::ModelProto> model_proto) : m_model_proto{model_proto} {
-    // Walk through the elements of opset_import field and register operator sets
-    // for each domain. An exception UnknownDomain() will raise if the domain is
-    // unknown or invalid.
-    for (const auto& id : m_model_proto->opset_import()) {
-        auto domain = id.has_domain() ? id.domain() : "";
-        if (domain == "ai.onnx") {
-            domain = "";
-        }
-        m_opset.emplace(domain, OperatorsBridge::get_operator_set(domain, id.version()));
+    // copy the opset imports from the ONNX model and sort them by their version in ascending order
+    // this will make sure that multiple opset imports for the same domain will cause the largest
+    // version to be used for this model, for example:
+    // [{domain:"", version:11}, {domain:"", version:1} {domain:"", version:13}] ==> {domain:"", version:13}
+    auto opset_imports = m_model_proto->opset_import();
+    const auto sort_by_version_ascending = [](const ONNX_NAMESPACE::OperatorSetIdProto& lhs,
+                                              const ONNX_NAMESPACE::OperatorSetIdProto& rhs) {
+        return lhs.version() < rhs.version();
+    };
+    std::sort(std::begin(opset_imports), std::end(opset_imports), sort_by_version_ascending);
+
+    for (const auto& id : opset_imports) {
+        const auto domain = id.has_domain() ? id.domain() == "ai.onnx" ? "" : id.domain() : "";
+        m_opset[domain] = OperatorsBridge::get_operator_set(domain, id.version());
     }
     // onnx.proto(.3): the empty string ("") for domain or absence of opset_import field
     // implies the operator set that is defined as part of the ONNX specification.
     const auto dm = m_opset.find("");
     if (dm == std::end(m_opset)) {
-        m_opset.emplace("", OperatorsBridge::get_operator_set("", ONNX_OPSET_VERSION));
+        m_opset[""] = OperatorsBridge::get_operator_set("", ONNX_OPSET_VERSION);
```

Comment:
Actually with `emplace` or `insert` there is no need to call `m_opset.find("")`, as they don't change the map if the key exists. Just wondered, is there any particular reason for this change?

Reply:
That's right, there's no reason. The if was there before and I've changed emplace to operator[] just as an alignment with the lines of code above. I can clean it up later in the scope of 76153.